govc/vm/putusbscancodes.go

```diff
@@ -0,0 +1,316 @@
+	"4": {0x21, false},
+	"5": {0x22, false},
+	"6": {0x23, false},
+	"7": {0x24, false},
+	"8": {0x25, false},
+	"9": {0x26, false},
+	"0": {0x27, false},
+	"A": {0x04, true},
+	"B": {0x05, true},
+	"C": {0x06, true},
+	"D": {0x07, true},
+	"E": {0x08, true},
+	"F": {0x09, true},
+	"G": {0x0a, true},
+	"H": {0x0b, true},
+	"I": {0x0c, true},
+	"J": {0x0d, true},
+	"K": {0x0e, true},
+	"L": {0x0f, true},
+	"M": {0x10, true},
+	"N": {0x11, true},
+	"O": {0x12, true},
+	"P": {0x13, true},
+	"Q": {0x14, true},
+	"R": {0x15, true},
+	"S": {0x16, true},
+	"T": {0x17, true},
+	"U": {0x18, true},
+	"V": {0x19, true},
+	"W": {0x1a, true},
+	"X": {0x1b, true},
+	"Y": {0x1c, true},
+	"Z": {0x1d, true},
+	"!": {0x1e, true},
+	"@": {0x1f, true},
+	"#": {0x20, true},
+	"$": {0x21, true},
+	"%": {0x22, true},
+	"^": {0x23, true},
+	"&": {0x24, true},
+	"*": {0x25, true},
+	"(": {0x26, true},
+	")": {0x27, true},
+	"-": {0x2d, false},
+	"_": {0x2d, true},
+	"=": {0x2e, false},
+	"+": {0x2e, true},
+	"[": {0x2f, false},
+	"{": {0x2f, true},
+	"]": {0x30, false},
+	"}": {0x30, true},
+	`\`: {0x31, false},
+	"|": {0x31, true},
+	";": {0x33, false},
+	":": {0x33, true},
+	"'": {0x34, false},
+	`"`: {0x34, true},
+	"`": {0x35, false},
+	"~": {0x35, true},
+	",": {0x36, false},
+	"<": {0x36, true},
+	".": {0x37, false},
+	">": {0x37, true},
+	"/": {0x38, false},
+	"?": {0x38, true},
+}
+
+type putusbscancode struct {
+	*flags.VirtualMachineFlag
+
+	UsbHidCodeValue int32
+	UsbHidCode      string
+	UsbHidString    string
+	LeftControl     bool
+	LeftShift       bool
+	LeftAlt         bool
+	LeftGui         bool
+	RightControl    bool
+	RightShift      bool
+	RightAlt        bool
+	RightGui        bool
+}
+
+func init() {
+	cli.Register("vm.putusbscancode", &putusbscancode{})
+}
+
+func (cmd *putusbscancode) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.VirtualMachineFlag, ctx = flags.NewVirtualMachineFlag(ctx)
+	cmd.VirtualMachineFlag.Register(ctx, f)
+
+	f.Var(flags.NewInt32(&cmd.UsbHidCodeValue), "r", "Raw USB HID Code Value (int32)")
+	f.StringVar(&cmd.UsbHidCode, "c", "", "USB HID Code (hex)")
+	f.StringVar(&cmd.UsbHidString, "s", "", "Raw String to Send")
+	f.BoolVar(&cmd.LeftControl, "lc", false, "Enable/Disable Left Control")
+	f.BoolVar(&cmd.LeftShift, "ls", false, "Enable/Disable Left Shift")
+	f.BoolVar(&cmd.LeftAlt, "la", false, "Enable/Disable Left Alt")
+	f.BoolVar(&cmd.LeftGui, "lg", false, "Enable/Disable Left Gui")
+	f.BoolVar(&cmd.RightControl, "rc", false, "Enable/Disable Right Control")
+	f.BoolVar(&cmd.RightShift, "rs", false, "Enable/Disable Right Shift")
+	f.BoolVar(&cmd.RightAlt, "ra", false, "Enable/Disable Right Alt")
+	f.BoolVar(&cmd.RightGui, "rg", false, "Enable/Disable Right Gui")
+}
+
+func (cmd *putusbscancode) Usage() string {
+	return ""
+}
+
+func (cmd *putusbscancode) Description() string {
+	return `Send Keystroke to VM.
+
+Examples:
+ Default Scenario
+  govc vm.putusbscancode $vm -r 1376263 (writes an 'r' to the console)
```

Comment:
Tried it out, the examples should be `-vm $vm`.  Or to support both, can do this (and add "VM" to Usage): https://github.com/vmware/govmomi/blob/cf359b62d1e6a76bcfeee0d5cc1bfb8f7db685b7/govc/vm/console.go#L82-L91